pkg/apis/sources/v1alpha1/interfaces.go

```diff
@@ -47,6 +47,17 @@ type EventSource interface {
 	AsEventSource() string
 }
 
+// multiTenant is implemented by all multi-tenant source types.
+type multiTenant interface {
+	IsMultiTenant() bool
+}
```

Comment:
This tiny interface is implemented by a source type to indicate that it should operate in multi-tenant mode.

In multi-tenant mode:
* there is only a single adapter deployed per namespace (named `<type>-adapter`)
* this adapter is owned by the service account with the same name, instead of by the source(s)
* the adapter's address gets a URL path appended in the form `/namespace/name`

The second point is because the service account is already owned by the source(s), so when all sources get deleted we can create a chain reaction of garbage collection service account -> adapter.
I'm fine reconsidering this design if someone has concerns.